tools/istio-iptables/pkg/dependencies/implementation_linux.go

```diff
@@ -76,67 +61,73 @@ func (r *RealDependencies) execute(cmd string, ignoreErrors bool, stdin io.Reade
 	return err
 }
 
-func (r *RealDependencies) executeXTables(cmd string, ignoreErrors bool, stdin io.ReadSeeker, args ...string) error {
-	log.Infof("Running command: %s %s", cmd, strings.Join(args, " "))
-
-	var stdout, stderr *bytes.Buffer
-	var err error
-	var nsContainer ns.NetNS
-
+func (r *RealDependencies) runCommand(c *exec.Cmd) error {
 	if r.CNIMode {
-		nsContainer, err = ns.GetNS(r.NetworkNamespace)
+		n, err := ns.GetNS(r.NetworkNamespace)
 		if err != nil {
 			return err
 		}
-		defer nsContainer.Close()
-	}
-	o := backoff.Option{
-		InitialInterval: 100 * time.Millisecond,
-		MaxInterval:     2 * time.Second,
+		defer n.Close()
+
+		return n.Do(func(ns.NetNS) error {
+			return c.Run()
+		})
 	}
```

Comment:
nit: IIRC we can simplify this a smidge with [`WithNetNSPath`](https://pkg.go.dev/github.com/containernetworking/plugins/pkg/ns#WithNetNSPath)